common/configuration/aws_secrets.go

```diff
@@ -0,0 +1,216 @@
+package configuration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"sync"
+
+	"github.com/TBD54566975/ftl/internal/slices"
+
+	"github.com/alecthomas/types/optional"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager/types"
+	"github.com/aws/smithy-go"
+)
+
+const schemeKey = "asm"
+
+// AWSSecrets implements Resolver and Provider for AWS Secrets Manager (ASM).
+//
+// The resolver does a direct/proxy map from a Ref to a URL, module.name <-> asm://module.name and does not access ASM at all.
+type AWSSecrets[R Role] struct {
+	AccessKeyID     string
+	SecretAccessKey string
+	Region          string
+	Endpoint        optional.Option[string]
+}
+
+var _ Resolver[Secrets] = AWSSecrets[Secrets]{}
+var _ Provider[Secrets] = AWSSecrets[Secrets]{}
+var _ MutableProvider[Secrets] = AWSSecrets[Secrets]{}
+
+var (
+	asmOnce   sync.Once
```

Comment:
Use [once.Once](https://pkg.go.dev/github.com/alecthomas/types@v0.16.0/once) instead, as it's both type and developer safe.

Reply:
Once isn't needed now with the constructor pattern.